src/fastfield/fast_value.rs

```diff
@@ -0,0 +1,241 @@
+use std::net::{IpAddr, Ipv6Addr};
+
+use crate::schema::{Cardinality, FieldType, Type};
+use crate::DateTime;
+pub fn ip_to_u128(ip_addr: IpAddr) -> u128 {
+    let ip_addr_v6: Ipv6Addr = match ip_addr {
+        IpAddr::V4(v4) => v4.to_ipv6_mapped(),
+        IpAddr::V6(v6) => v6,
+    };
+    u128::from_be_bytes(ip_addr_v6.octets())
+}
+
+/// Trait for large types that are allowed for fast fields: u128, IpAddr
+pub trait FastValueU128: Clone + Copy + Send + Sync + PartialOrd + 'static {
+    /// Converts a value from u128
+    ///
+    /// Internally all fast field values are encoded as u128.
+    /// **Note: To be used for converting encoded Term, Posting values.**
```

Comment:
We don't use it to convert Term / posting values do we?

Reply:
yep, copy paste error